frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -622,11 +658,19 @@ pub mod pallet {
 				Phase::Signed | Phase::Off
 					if remaining <= unsigned_deadline && remaining > Zero::zero() =>
 				{
-					// determine if followed by signed or not.
+					// our needs vary according to whether or not the unsigned phase follows a signed phase
 					let (need_snapshot, enabled, signed_weight) = if current_phase == Phase::Signed {
-						// followed by a signed phase: close the signed phase, no need for snapshot.
-						// TODO: proper weight https://github.com/paritytech/substrate/pull/7910.
-						(false, true, Weight::zero())
+						// there was previously a signed phase: close the signed phase, no need for snapshot.
+						//
+						// Notes:
+						//
+						//   - If the signed phase produced a viable solution, we disable the unsigned
+						//     phase. We want to prioritize signed solutions whenever they're available.
+						//   - `Self::finalize_signed_phase()` also appears in `fn do_elect`. This is
+						//     a guard against the case that `elect` is called prematurely. This adds
+						//     a small amount of overhead, but that is unfortunately unavoidable.
+						let (success, weight) = Self::finalize_signed_phase();
```

Comment:
the logic here is still correct, but the function can be named `try_finalize_signed` and return more meaningful stuff.

Reply:
I'm not sure. The most plausible alternate return values I can think of are `(Result<(), ()>, Weight)` and `Result<Weight, Weight>`. Neither of those is obviously better, to me.

Reply:
fair enough.